### PR TITLE
Fix spurious panic on bogus funding txn that confirm and are spent

### PR DESCRIPTION
**Pull request by TheBlueMatt**
In c02b6a3807488e1943d79792c5ac0ee52530b971 we moved the
`payment_preimage` copy from inside the macro which only runs if we
are spending an output we know is an HTLC output to doing it for
any script that matches our expected length. This can panic if an
inbound channel is created with a bogus funding transaction that
has a witness program of the HTLC-Success/-Offered length but which
does not have a second-to-last witness element which is 32 bytes.

Luckily this panic is relatively simple for downstream users to
work around - if an invalid-length-copy panic occurs, simply remove
the ChannelMonitor from the bogus channel on startup and run
without it. Because the channel must be funded by a bogus script in
order to reach this panic, the channel will already have closed by
the time the funding transaction is spent, and there can be no
local funds in such a channel, so removing the `ChannelMonitor`
wholesale is completely safe.

In order to test this we have to disable an in-line assertion that
checks that our transactions match expected scripts which we do by
checking for the specific bogus script that we now use in
`test_invalid_funding_tx`.

Thanks to Eugene Siegel for reporting this issue.

### CODE REVIEW

**Review by ariard — DISMISSED** [review]
Code Review ACK 9d91dac5, in conformity to what have been reviewed elsewhere.

**Review by ariard — COMMENTED** [review]
Code Review ACK dfc6a32c

Changes since last push :
- removing the use for `Transaction, TxOut, TxIn, OutPoint as BitcoinOutPoint` in functional tests
- removing the use for `Witness` in functional tests

**Review by ariard — APPROVED** [review]
[One More Time](https://www.youtube.com/watch?v=FGBhQbmPwH8) Code Review ACK 6c480ae :musical_note: 

> Oops, sorry, one more, looks like Vec::from(array) needed const generics which we don't support.

Hmmm, the previous branch built well on my local host. Some issue with bindings or other platforms than x86_64 ?

File: lightning/src/chain/channelmonitor.rs [hunk 1]
```diff
@@ -1729,6 +1729,26 @@ macro_rules! fail_unbroadcast_htlcs {
 	} }
 }
 
+// In the `test_invalid_funding_tx` test, we need a bogus script which matches the HTLC-Accepted
+// witness length match (ie is 136 bytes long). We generate one here which we also use in some
+// in-line tests later.
+
+#[cfg(test)]
+pub fn deliberately_bogus_accepted_htlc_witness_program() -> Vec<u8> {
```
Comment (dunxen) [inline lightning/src/chain/channelmonitor.rs:1737]: Looks all bogus to me, but perhaps the beginning of a new age techno rap.

**Review by dunxen — APPROVED** [review]
ACK 6c480ae

**Review by jkczyz — APPROVED** [review]

**Pull request merged**